run.go

```diff
@@ -326,15 +332,63 @@ func Log(c chan *SimulationStatus) DomainManipulator {
 	}
 }
 
-func (d *InMAP) checkOutputNames(n ...string) error {
+func removeDuplicates(s []string) []string {
+	result := make([]string, 0, len(s))
+	seen := make(map[string]string)
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			result = append(result, val)
+			seen[val] = val
+		}
+	}
+	return result
+}
+
+// checkForDerivitives creates a slice of unique model variables that are
+// required to calculate the requested outputVariables.
+func checkForDerivitives(m map[string]string, f map[string]govaluate.ExpressionFunction) (map[string]string, []string, error) {
+	getVariables := make([]string, 0, len(m))
+	for key, val := range m {
+		expression, err := govaluate.NewEvaluableExpressionWithFunctions(val, f)
+		if err != nil {
+			return nil, nil, fmt.Errorf("inmap: unsupported expression '%s'", val)
```

Comment:
We probably want to include the original error, unless there is a reason not to. Perhaps: 

return nil, nil, fmt.Errorf("inmap OutputVariables:  %v", err)